### PR TITLE
Note in README that substitutions don’t work inside quotes

### PR DESCRIPTION
**Pull request by havocp**
There’s more on this in HOCON.md but can’t hurt to repeat, people do miss it.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -706,9 +706,12 @@ string `foo` are concatenated into a string `42 foo`:
 When concatenating values into a string, leading and trailing
 whitespace is stripped but whitespace between values is kept.
 
-Unquoted strings also support substitutions of course:
+Quoted or unquoted strings can also concatenate with substitutions of course:
 
     tasks-url : ${base-url}/tasks
+    tasks-url : ${base-url}”tasks:colon-must-be-quoted”
```
Comment (TimMoore) [inline README.md]: It looks like some "smart" (curly) quotes snuck in.

Comment (TimMoore) [reply README.md]:
```suggestion
    tasks-url : ${base-url}"tasks:colon-must-be-quoted"
```

Comment (havocp) [reply README.md]: Thanks, apparently I shouldn't make PRs on an iPhone. Amusingly, the smart quotes appear to crash github's "commit suggestion" button so I have to fix this from my laptop ... 

image

**Comment by havocp** [comment]
@2m when I try to push another commit to this branch it says:
```
remote: error: GH006: Protected branch update failed for refs/heads/havocp-readme-quotes.
remote: error: Required status check "typesafe-cla-validator" is expected.
To github.com:lightbend/config.git
 ! [remote rejected] havocp-readme-quotes -> havocp-readme-quotes (protected branch hook declined)
error: failed to push some refs to 'git@github.com:lightbend/config.git'
```

Maybe I'm half asleep but I don't know what's going wrong there. I re-clicked the CLA to be sure but note the original PR has the CLA check passed.

**Comment by 2m** [comment]
Its the first time I've seen branch protection rule fail because of CLA validator **when** pushing an additional commit to the branch. Especially since CLA validator kicks in after the commit has been pushed.

To side-step this we could only set the `master` branch as the protected one. Currently all of the branches are protected in this repo.

**Comment by havocp** [comment]
Protecting only master is fine with me if you want to make that change (I don't think I have access). It seems like there's a bug here but not sure if it's in the cla-validator or in GitHub or something I'm doing wrong.

**Comment by 2m** [comment]
Alright, changed to only protect master.

**Pull request merged**